src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -242,16 +242,28 @@ private TextEdit[] MapEditsToHostDocument(RazorCodeDocument codeDocument, TextEd
             }
         }
 
-        private static bool IsIgnoredByCSharpFormatter(SyntaxNode? parent)
-            => parent?.AncestorsAndSelf().Any(n => n switch
+        private static bool ShouldIgnoreLineCompletely(SyntaxNode? parent, SourceText text)
+        {
+            return parent?.AncestorsAndSelf().Any(n => n switch
             {
                 // C# formatter doesn't touch array initializers
                 InitializerExpressionSyntax { RawKind: (int)CodeAnalysis.CSharp.SyntaxKind.ArrayInitializerExpression } => true,
                 // C# formatter doesn't touch object and collection initializers if they're not empty
                 InitializerExpressionSyntax { Expressions: { Count: > 0 } } => true,
+                // We don't want to format lines that are part of multi-line string literals
+                LiteralExpressionSyntax { RawKind: (int)CodeAnalysis.CSharp.SyntaxKind.StringLiteralExpression } => SpansMultipleLines(n, text),
```

Comment:
So, the goal here is to prevent formatting the multi-line literal? I think I get lost on why `AncestorsAndSelf` is the way to find line-mates.

Reply:
This ascends because it's saying "is the first thing on this line part of something that we don't want to format?"

But you raise an interesting point actually... What this is doing is checking what the first thing on a line is a child of, and with initializers the expressions in them could be arbitrarily complicated, so we have to ascend all the way. In the case of string literals it might not be necessary to ascend up the tree, we might be able to just check the parent directly. Will update and see if the tests still pass.